lib/spack/spack/cmd/ci.py

```diff
@@ -526,10 +547,8 @@ def ci_rebuild(args):
     if compiler_action != "FIND_ANY":
         install_args.append("--no-add")
 
-    # TODO: once we have the concrete spec registry, use the DAG hash
-    # to identify the spec to install, rather than the concrete spec
-    # json file.
-    install_args.extend(["-f", job_spec_json_path])
+    # Identify spec to install by hash
+    install_args.append("/{0}".format(job_spec.dag_hash()))
```

Comment:
how does this work with "install missing compilers"? then `job_spec` is neither in the db nor in the environment so it fails? or not?

Reply:
Unfortunately compiler bootstrapping pipelines haven't worked ever since we started using the `spack.lock` to communicate what to build to downstream jobs.